### PR TITLE
Run export correctly

### PR DESCRIPTION
**Pull request by hmaarrfk**
closes https://github.com/conda-forge/libitk-feedstock/issues/92

Once this is merged, we should issue cleanup PRs to the following recipes:
- https://github.com/conda-forge/vmtk-feedstock/blob/c37000b8cf9376914894339f0aa71a0e92df98b4/recipe/meta.yaml#L41
- https://github.com/conda-forge/libsimpleitk-feedstock/blob/13cd9c8a28d59abcdb5a32d1487e2dc99daa572e/recipe/meta.yaml#L33
- https://github.com/conda-forge/simpleitk-feedstock/blob/main/recipe/meta.yaml#L38
- https://github.com/conda-forge/ants-feedstock/blob/9b79e1577dacaccad744c17ff5cdaa98119822e6/recipe/meta.yaml#L24
- https://github.com/conda-forge/convert3d-feedstock/blob/1e4056c1d7a3052e70390017e358c77745473329/recipe/meta.yaml#L22
- https://github.com/conda-forge/petpvc-feedstock/blob/ff96acbb1a02fa95fd338d7d466c8fa14efd1605/recipe/meta.yaml#L23

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -57,6 +55,9 @@ outputs:
         - itkTestDriver -- echo "Runtime ITK OK"
         - itkTestDriver --no-process --compare-MD5 ITK-JPEG.jpg 959e1682d19026fc62cfbdc140df31da
   - name: libitk-devel
+    build:
+      run_exports:
+        - {{ pin_subpackage('libitk', max_pin='x.x') }}
```
Comment (effigies) [inline recipe/meta.yaml:60]: I think you also need a min_pin:

```suggestion
        - {{ pin_subpackage('libitk', min_pin='x.x', max_pin='x.x') }}
```

Comment (hmaarrfk) [reply recipe/meta.yaml:60]: no it takes it implicitly from the current package version.

**Comment by hmaarrfk** [comment]
Is there a reason to not cross compile aarch on linux? Things seem to work in https://github.com/conda-forge/libitk-feedstock/pull/94

**Comment by hmaarrfk** [comment]
because at compilation time, the user will have to depend on the `development` versions to do any kind of linking.

The non development cannot be used for linking at compile time, so shouldn't really have a run_export. 

See for example https://github.com/conda-forge/libsimpleitk-feedstock/blob/13cd9c8a28d59abcdb5a32d1487e2dc99daa572e/recipe/meta.yaml#L33 which only pull in the development package.

**Comment by blowekamp** [comment]
Thank you for improving the dependency specification. There have been lots of difficulties getting all the transitive dependency to work correctly. 

Also note that ITK is mostly a templated library. When a "patch" version is released dependent libraries likely should be recompiled to take advantage of the changes. Also I am uncertain how strict the ABI compatibility is between patch changes of the library. Perhaps the pinning should be "x.x.x"?


> See for example https://github.com/conda-forge/libsimpleitk-feedstock/blob/13cd9c8a28d59abcdb5a32d1487e2dc99daa572e/recipe/meta.yaml#L33 which only pull in the development package.

The above library is then used here: https://github.com/conda-forge/simpleitk-feedstock/blob/main/recipe/meta.yaml#L33
It took a long time to get the dependencies to work, much less be specified using the evolving conda features.

**Comment by hmaarrfk** [comment]
> Also note that ITK is mostly a templated library. When a "patch" version is released dependent libraries likely should be recompiled to take advantage of the changes. Also I am uncertain how strict the ABI compatibility is between patch changes of the library. Perhaps the pinning should be "x.x.x"?

I'm hesitant to pin so strictly since the compatibility should refer to "will the library link correctly".

It may be a really good note to add to the `meta.yaml` file, but I would like to keep the pinning looser rather than stricter (ITK is hard to compile as is, needlessly recompiling will be alot of churn).

Generally speaking, recompilation for libraries that use ITK is likely pretty frequent since they depend on other C dependencies that must be recompiled.

With the changes in this merge request:

https://github.com/conda-forge/simpleitk-feedstock/blob/main/recipe/meta.yaml#L38

the exact pinning should be removed and `libitk-devel` will enforce its own runtime requirements in accordance to the pinnings in this feedstock
image

This `exact` pinning is wrong (and likely causing you all headaches), and this PR will help remove it!

Thanks for the discussion!

**Review by blowekamp — APPROVED** [review]
LGTM

**Pull request merged**